### PR TITLE
[IMP] account, account_cancel, account_check_printing: The 'cancel' b…

### PR DESCRIPTION
**Pull request by hiralvbhavsar**
Description of the issue/feature this PR addresses:

Task: https://www.odoo.com/web?#id=28001&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
Pad: https://pad.odoo.com/p/r.ac0d12f6c9523ef7215d8ac115894d3c

Current behavior before PR:

 Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account_check_printing/models/account_payment.py [hunk 1]
```diff
@@ -123,7 +123,8 @@ def print_checks(self):
             }
         else:
             self.filtered(lambda r: r.state == 'draft').post()
-            self.write({'state': 'sent'})
+            if self.state != 'cancel':
+                self.write({'state': 'sent'})
```
Comment (mdi-odoo) [inline addons/account_check_printing/models/account_payment.py]: You can directly write like this: **self.filtered(lambda r: r.state != 'cancel').write({'state': 'sent'})**

File: addons/account/models/account_payment.py [hunk 2]
```diff
@@ -318,6 +318,10 @@ def cancel(self):
             rec.state = 'cancel'
 
     @api.multi
+    def action_set_to_draft(self):
+        self.state = 'draft'
```
Comment (mdi-odoo) [inline addons/account/models/account_payment.py]: Why you need to add this method. There is already a method **action_draft** that do the same thing.

File: addons/account/views/account_payment_view.xml [hunk 3]
```diff
@@ -76,8 +76,8 @@
                 <form string="Register Payment" version="7">
                     <header>
                         <button name="post" class="oe_highlight" states="draft" string="Confirm" type="object"/>
-                        <button name="action_draft" states="cancel" string="Set to Draft" type="object"/>
-                        <field name="state" widget="statusbar" statusbar_visible="posted,reconciled"/>
+                        <button name="action_set_to_draft" class="oe_highlight" states="cancel" string="Set To Draft" type="object"/>
```
Comment (mdi-odoo) [inline addons/account/views/account_payment_view.xml]: Why this change?

File: addons/account/views/account_payment_view.xml [hunk 4]
```diff
@@ -76,8 +76,8 @@
                 <form string="Register Payment" version="7">
                     <header>
                         <button name="post" class="oe_highlight" states="draft" string="Confirm" type="object"/>
-                        <button name="action_draft" states="cancel" string="Set to Draft" type="object"/>
-                        <field name="state" widget="statusbar" statusbar_visible="posted,reconciled"/>
+                        <button name="action_draft" class="oe_highlight" states="cancel" string="Set To Draft" type="object"/>
+                        <field name="state" widget="statusbar" statusbar_visible="draft,posted,reconciled,cancel"/>
```
Comment (smetl) [inline addons/account/views/account_payment_view.xml]: cancelled instead of cancel?

File: addons/account_cancel/views/account_views.xml [hunk 5]
```diff
@@ -65,7 +65,7 @@
             <field name="inherit_id" ref="account.view_account_payment_form"/>
             <field name="arch" type="xml">
                 <xpath expr="//header/field[@name='state']" position="before">
-                    <button name="cancel" states="posted,sent,reconciled" string="Cancel" type="object"/>
+                    <button name="cancel" attrs="{'invisible': ['|', ('state', '=', 'draft'), ('state', '=', 'cancel')]}" string="Cancel" type="object"/>
```
Comment (smetl) [inline addons/account_cancel/views/account_views.xml]: `[('state', 'in', ['draft', 'cancel'])]`

File: addons/account_check_printing/models/account_payment.py [hunk 6]
```diff
@@ -123,7 +123,7 @@ def print_checks(self):
             }
         else:
             self.filtered(lambda r: r.state == 'draft').post()
-            self.write({'state': 'sent'})
+            self.filtered(lambda r: r.state != 'cancel').write({'state': 'sent'})
```
Comment (smetl) [inline addons/account_check_printing/models/account_payment.py]: Maybe I'm wrong but is it not better to filter the cancelled payments at the beginning of the method?

File: addons/account/views/account_payment_view.xml (hunk 4 above)
Comment (hbh-odoo) [inline addons/account/views/account_payment_view.xml]: But for that, key is 'cancel' not 'cancelled'.

Comment (smetl) [inline addons/account/views/account_payment_view.xml]: I meant changing the key from 'cancel' to 'cancelled'
